mllib/src/main/scala/org/apache/spark/ml/classification/NaiveBayes.scala

```diff
@@ -247,12 +246,12 @@ class NaiveBayes @Since("1.5.0") (
     $(modelType) match {
       case Multinomial | Bernoulli =>
         val theta = new DenseMatrix(numLabels, numFeatures, thetaArray, true)
-        new NaiveBayesModel(uid, pi.compressed, theta.compressed, null)
+        new NaiveBayesModel(uid, pi.compressed, theta.compressed, Matrices.zeros(0, 0))
```

Comment:
Hm, but isn't it redundant to serialize all these zeroes? how big will it be -- O(n^2) if n is the number of features?
Is it backwards-compatible?

Reply:
In MNB/CNB/BNB, it is a empty matrix (numRows=0, numCols=0), so it should be tiny.
In GNB, it is a matrix of size = #class x #feature , as big as `theta` matrix.
I also test the backwards-compatibilty: save a model in 2.4.4, and load it in this PR, it works fine.